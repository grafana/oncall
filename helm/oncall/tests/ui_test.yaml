suite: test ui dev container
templates:
  - ui/deployment.yaml
release:
  name: oncall
tests:
  - it: doesn't create deployment if ui.enabled is false
    set:
      ui:
        enabled: false
    asserts:
      - hasDocuments:
          count: 0
  - it: creates deployment if ui.enabled is true
    set:
      ui:
        enabled: true
    asserts:
      - matchSnapshot: {}
  - it: doesn't add env vars if none are supplied
    set:
      ui:
        enabled: true
    asserts:
      - isEmpty:
          path: spec.template.spec.containers[0].env
  - it: adds env vars if they are supplied
    set:
      ui:
        enabled: true
        env:
          FOO: bar
          ABC: 123
    asserts:
      - equal:
          path: spec.template.spec.containers[0].env
          value:
            - name: ABC
              value: "123"
            - name: FOO
              value: bar
