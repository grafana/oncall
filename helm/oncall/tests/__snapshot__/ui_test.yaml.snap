creates deployment if ui.enabled is true:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: oncall-ui
      name: oncall-ui
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/component: oncall-ui
      template:
        metadata:
          labels:
            app.kubernetes.io/component: oncall-ui
        spec:
          containers:
            - env: null
              image: oncall/ui:dev
              imagePullPolicy: IfNotPresent
              name: oncall-ui
              volumeMounts:
                - mountPath: /etc/app
                  name: hot-reloaded-plugin
          volumes:
            - hostPath:
                path: /oncall-plugin
              name: hot-reloaded-plugin
